Java-Basics.md

```diff
@@ -0,0 +1,442 @@
+
+Objects are _instances_ of a class. All objects are instances of a certain class. Imagine a class being a "template", which every Object copies to. When you create an Object, basically it creates a new object on the blueprint of a class. Now lets look at this from a little piece of code :
+
+```java
+// Car class
+public class Car {
+  // car name
+  private String name;
+	// car mannufacturer name
+	private String manufacturerName;
+  // Constructor
+	public Car(String name, String man) {
+		this.name = name;
+		this.manufacturerName = man;
+	}
+  // Getter method
+	public String getName() {
+		return name;
+	}
+  // Getter method
+	public String getManufacturerName() {
+		return manufacturerName;
+	}
+}
+
+Car modelS = new Car("Model S","Tesla");
+
+System.out.println("Full Car Name = " + modelS.getManufacturerName() + " " + modelS.getName());
+// prints Tesla Model S
+```
+
+So, `Car` is a Class, which has the fields or properties name and manufacturerName. `modelS` is an object of `Car` Class. So `modelS` also has the same properties and methods.
+
+### Access Modifiers
+
+Ok, so if `modelS` has all the same properties, I should be able to access `name` or `manufacturerName` right?
+
+```java
+System.out.println(modelS.name);
+```
+Results in :
+
+```asciidoc
+Main.java:13: error: name has private access in Car
+		System.out.println(modelS.name);
+		                         ^
+1 error
+```
+
+Why? Notice we had mentioned `private` before the variable `name` in class `Car`. So since its private, we cant touch it from outside!
+
+There are other kinds of access Modifiers such as public, protected, default etc. [Read more about them here](#TODO).
+
+### Constructors
+
+What's the point then? I should be able to store data in it right?
+
+Thats when we use either Getter / Setter methods or in this case Constructors to initialize a class. Basically every Java Class has a constructor, which is the method which is called first when any object of the class is initialized. Think of it as a bit of starter code.
+
+When you write a Class without any constructor, then Java assumes it has a default constructor :
+
+```java
+
+public class Car {
+	private String name;
+}
+
+Car modelS = new Car();
+```
+
+This initializing with no parameters is a way of calling the default constructor. You can also have a default constructor written yourself this way :
+
+```java
+public class Car {
+	private String name;
+
+	public Car() {
+		name = "Tesla";
+	}
+}
+```
+
+Then, when calling `new Car()`, the variable `name` will get auto-initialized to `"Tesla"`.
+
+Like what Constructors can do? [Dive into it more here](#TODO).
+
+### Methods
+
+`getName()` and `getManufacturerName()` are two "Getter" methods we have used here. Notice unlike JavaScript, we **have** to define the return type of any method we write, otherwise it will fail at compile time. If you do not want a method to return anything, use `void` return type.
```

Comment:
Link this warning with the info at the top, that Java is a strongly typed language.